### DISCUSSION TITLE
Environment.NewLine instead of \n in C#

### DISCUSSION THREAD

**Discussion by ShervanN** [ProfessionalCSharp/ProfessionalCSharp2021#122] [Q&A] [answered]
Hi,

Why do you use **Environment.NewLine** property instead of **\n** in your code? 
On page 295, Using Combinators.

> private async static void MultipleAsyncMethodsWithCombinators1()
> {
> Task<string> t1 = GreetingAsync("Stephanie");
> Task<string> t2 = GreetingAsync("Matthias");
> await Task.WhenAll(t1, t2);
> Console.WriteLine($"Finished both methods.{**Environment.NewLine**} " +
> $"Result 1: {t1.Result}{Environment.NewLine} Result 2: {t2.Result}");
> }

**Comment by christiannagel** [comment, marked as answer]
Hello Shervan, thanks for asking!

The new line character is different based on the platform. See this [Wikipedia article - newline](https://en.wikipedia.org/wiki/Newline).

`Environment.NewLine` returns `\r\n` with Windows, and `\n` with Linux.
